nc/source/graphics/vulkan/TextureManager.cpp

```diff
@@ -0,0 +1,147 @@
+#include "graphics/vulkan/TextureManager.h"
+#include "graphics/vulkan/Initializers.h"
+#include "graphics/vulkan/resources/ResourceManager.h"
+#include "graphics/vulkan/resources/GraphicsResources.h"
+#include "debug/Utils.h"
+#include "config/Config.h"
+#include "config/ConfigInternal.h"
+
+#define STB_IMAGE_IMPLEMENTATION
+#include "stb/stb_image.h"
+
+namespace
+{
+    using namespace nc::graphics::vulkan;
+    TextureManager* impl = nullptr;
+} // end anonymous namespace
+
+namespace nc::graphics::vulkan
+{
+    void LoadTextures(const std::vector<std::string>& paths)
+    {
+        IF_THROW(!impl, "graphics::vulkan::LoadMeshAsset - impl is not set");
+        impl->LoadTextures(paths);
+    }
+    
+    TextureManager::TextureManager(nc::graphics::Graphics2* graphics)
+    : m_graphics{graphics}
+    {
+        impl = this;
+    }
+
+    void TextureManager::LoadTextures(const std::vector<std::string>& paths)
+    {
+        TexturesData textures{};
+
+        textures.sampler = m_graphics->GetBasePtr()->CreateTextureSampler();
+        textures.layout = vk::ImageLayout::eShaderReadOnlyOptimal;
+
+        uint32_t index = 0;
+        for (auto& path : paths)
+        {
+            if (ResourceManager::TextureExists(path)) continue;
+            
+            int32_t width, height, numChannels;
+
+            stbi_uc* pixels = stbi_load(path.c_str(), &width, &height, &numChannels, STBI_rgb_alpha);
+
+            if (!pixels)
+            {
+                throw std::runtime_error("Failed to load texture file: " + path);
+            }
+
+            auto image = std::make_unique<ImmutableImage>();
```

Comment:
These three lines make me think ImmutableImage should have another constructor that does what Bind does. Then this would look like:

textureBuffers.push_back(std::make_unique<ImmutableImage>(m_graphics, pixels, width, height));

If what we talked about with ResourceManager result in textureBuffers not having uniquew pointers, even better:

textureBuffers.emplace_back(m_graphics, pixels, width, height);